command/agent/agent_endpoint_test.go

```diff
@@ -446,6 +447,141 @@ func TestHTTP_AgentMonitor(t *testing.T) {
 	})
 }
 
+func TestHTTP_AgentMonitorExport(t *testing.T) {
+	ci.Parallel(t)
+	const expectedText = "log log log log log"
+
+	testFile, err := os.CreateTemp("", "nomadtests")
+	must.NoError(t, err)
+
+	_, err = testFile.Write([]byte(expectedText))
+	must.NoError(t, err)
+	inlineFilePath := testFile.Name()
+	config := func(c *Config) {
+		c.LogFile = inlineFilePath
+	}
+	baseURL := "/v1/agent/monitor/export?"
+	cases := []struct {
+		name        string
+		follow      string
+		nodeID      string
+		onDisk      string
+		serviceName string
+		serverID    string
+
+		errCode       int
+		expectErr     bool
+		nomadFilePath string
+		want          string
+	}{
+		{
+			name:   "happy_path",
+			follow: "false",
+			onDisk: "true",
+
+			expectErr:     false,
+			nomadFilePath: inlineFilePath,
+			want:          expectedText,
+		},
+		{
+			name:   "invalid_onDisk",
+			follow: "false",
+			onDisk: "green",
+
+			errCode:       400,
+			expectErr:     true,
+			nomadFilePath: inlineFilePath,
+			want:          expectedText,
+		},
+		{
+			name:   "invalid_follow",
+			follow: "green",
+			onDisk: "false",
+
+			errCode:       400,
+			expectErr:     true,
+			nomadFilePath: inlineFilePath,
+			want:          expectedText,
+		},
+
+		{
+			name:     "server_and_node",
+			follow:   "false",
+			onDisk:   "true",
+			nodeID:   "doesn'tneedtobeuuid",
+			serverID: "doesntneedtobeuuid",
+
+			errCode:       400,
+			expectErr:     true,
+			nomadFilePath: inlineFilePath,
+			want:          expectedText,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			httpTest(t, config, func(s *TestAgent) {
+
+				urlString := baseURL +
+					"on_disk=" + tc.onDisk +
+					"&service_name=" + tc.serviceName +
+					"&follow=" + tc.follow +
+					"&node_id=" + tc.nodeID +
+					"&server_id=" + tc.serverID +
+					"&mocked=" + "true"
+
+				req, err := http.NewRequest(http.MethodGet, urlString, nil)
+				must.NoError(t, err)
+
+				resp := newClosableRecorder()
+				defer resp.Close()
+				var (
+					builder strings.Builder
+					frame   sframer.StreamFrame
+					wg      sync.WaitGroup
+				)
+				errCh := make(chan error, 1)
+				wg.Add(1)
+				go func(errCh chan error) {
+					defer wg.Done()
+
+					_, err = s.Server.AgentMonitorExport(resp, req)
+					if err != nil {
+						errCh <- err
+					}
+
+				}(errCh)
+				wg.Wait()
+				select {
+				case err := <-errCh:
+					if tc.expectErr {
+						must.Eq(t, err.(HTTPCodedError).Code(), tc.errCode)
+						return
+					} else {
+						must.Unreachable(t)
```

Comment:
"Unreachable" is totally correct, but this will report the error message that we shouldn't be getting:

```suggestion
						must.NoError(t, err)
```